File: tests/memory/fixtures/tiny_repo/py_pkg/greeter.py
"""Greeter module — one class, one method, one function."""

from typing import Union

Name = Union[str, bytes]


class Greeter:
    """A polite greeter."""

    def greet(self, who: Name) -> str:
        """Return a greeting for ``who``."""
        return f"Hello, {who!r}"


def make_greeter() -> Greeter:
    """Factory that returns a fresh Greeter."""
    return Greeter()

File: tests/memory/fixtures/tiny_repo/ts_pkg/greeter.ts
/** A polite greeter written in TypeScript. */
export class Greeter {
  greet(name: string): string {
    return `Hello, ${name}`;
  }
}

/** Factory that returns a fresh Greeter. */
export function makeGreeter(): Greeter {
  return new Greeter();
}

/** Arrow function assigned to a const at module scope. */
export const shout = (name: string): string => `HELLO, ${name.toUpperCase()}`;
